### PR TITLE
Leverage the babel rest-object spread plugin #304

### LINKED ISSUE #304: Leverage the babel rest-object spread plugin
In order to avoid the verbosity of `Object.assign`, we should use [ES6 Object Rest/Spread Properties](https://github.com/sebmarkbage/ecmascript-rest-spread).

So instead of writing things like

```js
const updated = Object.assign({}, initial, {_status: "synced"});
```

We'd get

```js
const updated = {...initial, _status: "synced"};
```

This implies enabling the [object rest spread babel plugin](https://www.npmjs.com/package/babel-plugin-transform-object-rest-spread).

### PR DESCRIPTION
**Pull request by lavish205**
Fixes #304

### CODE REVIEW

**Comment by Natim** [comment]
Please rebase on top of master.

File: fx-src/index.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ export default function loadKinto() {
         events: emitter
       };
 
-      const expandedOptions = Object.assign(defaults, options);
+      const expandedOptions = {...defaults, options};
```
Comment (leplatrem) [inline fx-src/index.js]: I think in this case it should be `{...defaults, ...options}`

File: src/collection.js [hunk 2]
```diff
@@ -59,6 +59,7 @@ export class SyncResultObject {
      */
     this.ok = true;
     Object.assign(this, SyncResultObject.defaults);
+    ///{...this, ...SyncResultObject.defaults};
```
Comment (leplatrem) [inline src/collection.js]: leftover?

File: src/collection.js [hunk 3]
```diff
@@ -814,7 +815,9 @@ export default class Collection {
         // Merge outgoing conflicts into sync result object
         syncResultObject.add("conflicts", conflicts);
         // Reflect publication results locally
-        const missingRemotely = skipped.map(r => Object.assign({}, r, {deleted: true}));
+        const missingRemotely = skipped.map(r => {
+          return {...r, deleted: true};
+        });
```
Comment (leplatrem) [inline src/collection.js]: nitpick: you can write this on one line:
```js
const missingRemotely = skipped.map(r => ({...r, deleted: true}));
```

File: src/index.js [hunk 4]
```diff
@@ -29,7 +29,7 @@ export default class Kinto extends KintoBase {
       events: new EventEmitter()
     };
 
-    super(Object.assign({}, defaults, options));
+    super({...defaults, options});
```
Comment (leplatrem) [inline src/index.js]: in this case it's `{...defaults, ...options}`

File: src/KintoBase.js [hunk 5]
```diff
@@ -55,7 +55,7 @@ export default class KintoBase {
       bucket: DEFAULT_BUCKET_NAME,
       remote: DEFAULT_REMOTE,
     };
-    this._options = Object.assign(defaults, options);
+    this._options = {...defaults, options};
```
Comment (leplatrem) [inline src/KintoBase.js]: I think in this case it should be `{...defaults, ...options}`

**Pull request closed without merging**